src/main/scala/firrtl/EmissionOption.scala

```diff
@@ -0,0 +1,63 @@
+// See LICENSE for license details.
+
+package firrtl
+
+/** 
+  * Base type for emission customization options 
+  * NOTE: all the following traits must be mixed with SingleTargetAnnotation[T <: Named]
+  *       in order to be taken into account in the Emitter
+  */
+trait EmissionOption
```

Comment:
You can use a self-type to force the `NOTE` requirement above:
```suggestion
trait EmissionOption { this: SingleTargetAnnotation[_] => }
```

Reply:
There's also kind of an implication that each of these should be annotating specific things. Maybe instead of traits with mixins we should just make stand alone annotations that have these fields.

Reply:
@seldridge actually I can't since I inherit them with case objects that are pretty convienient